src/vehicle.cpp

```diff
@@ -3193,6 +3205,43 @@ int vehicle::basic_consumption(const itype_id &ftype) const
     return fcon;
 }
 
+int vehicle::supplemental_consumption( const itype_id &ftype ) const
+{
+    int fcon = 0;
+    if( chainsaw_on ) {
+        std::vector<int> all_chainsaws = all_parts_with_feature("CHAINSAW");
+        for( size_t c = 0; c < all_chainsaws.size(); ++c ) {
+            if( part_info( all_chainsaws[c] ).fuel_type == ftype ) {
+                fcon += part_power( all_chainsaws[c] );
+                if( one_in( 8 ) ) {
+                    tripoint part_pos = global_pos3() + parts[ all_chainsaws[c] ].precalc[0];
+                    //~ sound of working chainsaw
+                    sounds::sound(part_pos, 20, _( "vrr-vrr-vroom!" ));
```

Comment:
And yet, each of my offerings has more character than vrr-vrr-vroom, which
seems more like the sound I'd make playing with hot wheels... Dignity is
overrated, have fun with it.
On Dec 27, 2015 11:18 PM, "Angela Graves" <notifications@github.com> wrote:

> In src/vehicle.cpp
> <https://github.com/CleverRaven/Cataclysm-DDA/pull/14639#discussion_r48463344>
> :
>
> > @@ -3193,6 +3205,43 @@ int vehicle::basic_consumption(const itype_id &ftype) const
> >      return fcon;
> >  }
> >
> > +int vehicle::supplemental_consumption( const itype_id &ftype ) const
> > +{
> > +    int fcon = 0;
> > +    if( chainsaw_on ) {
> > +        std::vector<int> all_chainsaws = all_parts_with_feature("CHAINSAW");
> > +        for( size_t c = 0; c < all_chainsaws.size(); ++c ) {
> > +            if( part_info( all_chainsaws[c] ).fuel_type == ftype ) {
> > +                fcon += part_power( all_chainsaws[c] );
> > +                if( one_in( 8 ) ) {
> > +                    tripoint part_pos = global_pos3() + parts[ all_chainsaws[c] ].precalc[0];
> > +                    //~ sound of working chainsaw
> > +                    sounds::sound(part_pos, 20, _( "vrr-vrr-vroom!" ));
>
> Yeah, like that.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/CleverRaven/Cataclysm-DDA/pull/14639/files#r48463344>.
>